app/models/container.js

```diff
@@ -16,6 +16,9 @@ function ContainerViewModel(data) {
   for (var i = 0; i < data.Ports.length; ++i) {
     var p = data.Ports[i];
     if (p.PublicPort) {
+      if (undefined !== currentEndpoint && currentEndpoint.URLPublish !== '') {
```

Comment:
This should not be located inside the `ContainerViewModel` object. It should not be coupled with endpoint data.